#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/Youbot.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/mir/mir100/protos/Mir100UltrasonicSensor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/apartment_structure/protos/Wall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Roughcast.proto"

WorldInfo {
  contactProperties [
    ContactProperties {
      material1 "InteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation -0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
    ContactProperties {
      material1 "InteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation 0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
  ]
}
Viewpoint {
  orientation 0.08509996602572453 0.9125938670534115 -0.39991302754339086 0.526495997939232
  position -6.832774814657631 0.023425095672641394 3.7091356797206463
  followType "Pan and Tilt Shot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 0 0.05
  name "rectangle arena(1)"
  floorSize 10 10
  floorAppearance Appearance {
    texture ImageTexture {
      url [
        "../../../Battle area3.png"
      ]
    }
    textureTransform TextureTransform {
      scale 0.05 0.05
    }
  }
}
Solid {
  translation -1.17 -0.85 0.051
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
          emissiveColor 1 0 0
          specularColor 1 0 0
        }
      }
      geometry Plane {
        size 0.025 0.025
      }
    }
  ]
  name "Red"
}
Solid {
  translation -1.17 -1.05 0.051
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 1
          emissiveColor 0 0 1
          specularColor 0 0 1
        }
      }
      geometry Plane {
        size 0.025 0.025
      }
    }
  ]
  name "Blue"
}
Solid {
  translation -1.17 -0.51 0.051
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 1 0
          emissiveColor 0 1 0
          specularColor 0 1 0
        }
      }
      geometry Plane {
        size 0.025 0.025
      }
    }
  ]
  name "Green"
}
Solid {
  translation -1.17 -0.66 0.051
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 0
          emissiveColor 1 1 0
          specularColor 1 1 0
        }
      }
      geometry Plane {
        size 0.025 0.025
      }
    }
  ]
  name "Yellow"
}
Youbot {
  hidden translation_30 0 0 0.077
  hidden rotation_30 0 0 1 5.405651672845218e-07
  hidden translation_31 0.033 0 0.07
  hidden rotation_31 0 0.9999999999999999 0 2.1900144899933154e-07
  hidden translation_32 0 0 0.155
  hidden rotation_32 0 1 0 3.39798755973321e-07
  hidden translation_33 0 0 0.135
  hidden rotation_33 0 -1 0 0
  hidden translation_34 0 0 0.081
  hidden rotation_34 0 0 1 0
  hidden translation_35 0 5.104939011827844e-10 0.09
  hidden rotation_35 1 0 0 4.71239
  hidden translation_36 0 -4.75461482316335e-10 0.09
  hidden rotation_36 0 -0.7071067811865476 0.7071067811865476 3.14159
  hidden translation_37 0.228 -0.158 -0.055
  hidden rotation_37 0 1 0 0.33301009334676973
  hidden translation_38 0.228 0.158 -0.055
  hidden rotation_38 0 1 0 2.21336164913174
  hidden translation_39 -0.228 -0.158 -0.055
  hidden rotation_39 0 1 0 0.3427633480192099
  hidden translation_40 -0.228 0.158 -0.055
  hidden rotation_40 0 1 0 2.5790955200056436
  translation -1.14129 -2.09477 0.155367
  rotation -0.0021625515651156023 0.0020567677300412615 0.9999955465286998 1.5819113004328902
  name "KOKI2"
  controller "line_follow_v1"
  bodySlot [
    DistanceSensor {
      translation 0.23 0.01 -0.04
      name "if1"
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
    }
    DistanceSensor {
      translation 0.23 0 -0.04
      name "if2"
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
    }
    DistanceSensor {
      translation 0.23 0.1 -0.04
      name "if3"
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
    }
    DistanceSensor {
      translation 0.13 0.14 -0.02
      rotation 0 0 1 1.5708
      name "if4"
      lookupTable [
        0 0 0
        0.8 1000 0
      ]
    }
    DistanceSensor {
      translation 0.02 0.14 -0.02
      rotation 0 0 1 1.5708
      name "if5"
      lookupTable [
        0 0 0
        0.8 1000 0
      ]
    }
    DistanceSensor {
      translation -0.1 0.14 -0.02
      rotation 0 0 1 1.5708
      name "if6"
      lookupTable [
        0 0 0
        0.8 1000 0
      ]
    }
    DistanceSensor {
      translation -0.23 0.1 -0.02
      rotation 0 0 1 3.14159
      name "if7"
    }
    DistanceSensor {
      translation -0.23 0 -0.02
      rotation 0 0 1 3.14159
      name "if8"
    }
    DistanceSensor {
      translation -0.23 -0.1 -0.02
      rotation 0 0 1 3.14159
      name "if9"
    }
    DistanceSensor {
      translation -0.1 -0.14 -0.03
      rotation 0 0 1 -1.5708
      name "if10"
      lookupTable [
        0 0 0
        0.8 1000 0
      ]
    }
    DistanceSensor {
      translation 0.02 -0.14 -0.03
      rotation 0 0 1 -1.5708
      name "if11"
      lookupTable [
        0 0 0
        0.8 1000 0
      ]
    }
    DistanceSensor {
      translation 0.13 -0.14 -0.03
      rotation 0 0 1 -1.5708
      name "if12"
      lookupTable [
        0 0 0
        0.8 1000 0
      ]
    }
    DistanceSensor {
      translation 0.23 0.1 -0.03
      rotation 0 0 1 0.7854000000000001
      name "if13"
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
    }
    Lidar {
      translation 0.27 0 -0.09
      numberOfLayers 1
    }
    Mir100UltrasonicSensor {
      translation 0.28 -0.01 -0.074
      name "ultrasonic_sensor"
      lookupTable [
        0.01 995 0
        0.35 0 0
      ]
    }
    Camera {
      translation 0.28 -0.01 -0.0799993
      rotation 0 1 0 1.5708
      fieldOfView 0.8
      recognition Recognition {
        frameThickness 3
      }
    }
    DistanceSensor {
      translation 0.266 0.11 -0.0838586
      rotation 0 1 0 1.507
      name "ir1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.266 -0.11 -0.0838586
      rotation 0 1 0 1.507
      name "ir12"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.268 0.095 -0.0870525
      rotation 0 1 0 1.507
      name "ir2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.268 -0.095 -0.0870525
      rotation 0 1 0 1.507
      name "ir11"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.271 0.078 -0.0870461
      rotation 0 1 0 1.507
      name "ir3"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.271 -0.078 -0.0870461
      rotation 0 1 0 1.507
      name "ir10"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.2772 0.066 -0.0864123
      rotation 0 1 0 1.507
      name "ir4"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.2808 0.05 -0.0864123
      rotation 0 1 0 1.507
      name "ir5"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.2817 0.025 -0.0864086
      rotation 0 1 0 1.507
      name "ir6"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.2818 -0.004 -0.0857775
      rotation 0 1 0 1.507
      name "ir7"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.282 -0.034 -0.0870461
      rotation 0 1 0 1.507
      name "ir8"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.276 -0.066 -0.0876836
      rotation 0 1 0 1.507
      name "ir9"
      type "infra-red"
    }
  ]
}
Wall {
  translation 2.4 3.26 0.1
  name "wall(5)"
  size 0.38 1 0.4
  appearance Roughcast {
    colorOverride 0 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -3.4 3.78 0.1
  name "wall(6)"
  size 0.38 1 0.4
  appearance Roughcast {
    colorOverride 0 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -3.35 1.5 0.1
  name "wall(7)"
  size 0.38 1 0.4
  appearance Roughcast {
    colorOverride 0 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -3.38 -4.29 0.1
  name "wall(1)"
  size 0.38 1 0.4
  appearance Roughcast {
    colorOverride 0 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
